src/data_node/data_node.py

```diff
@@ -27,12 +28,33 @@ def __init__(
         self.name_node_stub = name_node_pb2_grpc.NameNodeServiceStub(
             self.name_node_channel)
 
-    def SendFile(self, request_iterator, context):
-        filename = os.path.join(self.dir, 'test1.txt')
-        SaveChunksToFile(request_iterator, filename)
-        file_size = GetFileSize(filename)
+    def SendFile(self, request, context):
+        current_dir = os.path.dirname(os.path.abspath(__file__)) 
+        
+        storage_base_dir = os.path.join(current_dir,'storage')
+
+        user_dir = os.path.join(storage_base_dir, request.username)
+        os.makedirs(user_dir, exist_ok=True) 
+
+        file_dir = os.path.join(user_dir, request.filename)
+        os.makedirs(file_dir, exist_ok=True)  
+
+        chunk_file_name = f"chunk_{request.chunk_number}"
+        chunk_file_path = os.path.join(file_dir, chunk_file_name)
+
+        with open(chunk_file_path, 'wb') as f:
+            f.write(request.chunk_data)
+
+        chunk_size_MB = len(request.chunk_data) / MB_IN_BYTES
+        print(f"Received chunk {request.chunk_number} of size {chunk_size_MB} MB")
+        print(f"Capacity before: {self.capacity_MB} MB")
+        self.capacity_MB -= chunk_size_MB
+        print(f"Capacity after: {self.capacity_MB} MB")
+        
+        file_size = os.path.getsize(chunk_file_path)
         return data_node_pb2.Reply(length=file_size)
```

Comment:
In other functions, the return statement has one space between itself and the line written before.